### ISSUE TITLE
NAN loss problem[SOLVED]: Performance Plummets When Setting adj_label Diagonal to Zero

### ISSUE THREAD

**Issue by ChildishBob** [kickingass/Graph-MLP#1]
After setting the diagonal of adj_label to zero:
![image](https://user-images.githubusercontent.com/56781141/124377719-f91a4c00-dcdf-11eb-8ec3-002bfe45e9f6.png)

The test accuracy for cora goes down to 0.13 with the same configuration:
![image](https://user-images.githubusercontent.com/56781141/124377795-6f1eb300-dce0-11eb-8592-25f7e7a9bbc5.png)

which is quite confusing since this modification is in accordance with:
![image](https://user-images.githubusercontent.com/56781141/124377857-d6d4fe00-dce0-11eb-9b4e-0c5492ea1af3.png)

**Comment by yanghu819** [comment]
Thank you so much for your issue on the NAN loss in training：

We fix the bug, for the code will bring NAN loss in training because "log(0)" in 
![image](https://user-images.githubusercontent.com/44990984/124384103-6f2faa80-dd02-11eb-8dca-30e95adbfaa5.png)

We add 1e-8 to prevent the NAN loss. Now the code is updated, please try it.
-------------
Also, you don't have to add the code for we remove the distance of its self in get_feature_dis in models.py
![image](https://user-images.githubusercontent.com/44990984/124384189-adc56500-dd02-11eb-9de4-29a6eb53a492.png)

**Comment by ChildishBob** [comment]
Thanks for your work! By the way, you said
> Also, you don't have to add the code for we remove the distance of its self in get_feature_dis in models.py

However, in
![image](https://user-images.githubusercontent.com/56781141/124404969-b99b4080-dd6f-11eb-8ba4-b48dc8ec8537.png)

even when x_dis is 0, we have e^0 = 1. That conflicts with 
![image](https://user-images.githubusercontent.com/56781141/124404891-6de89700-dd6f-11eb-9ffb-93fd31cd6058.png)

**Comment by yanghu819** [comment]
I think you did a very good job in finding this minute difference. That indeed will result a bit difference in formula，but the loss function optimized won' t be influenced by this operation: 
an added constant value will not influence the gradient much, also, you can try to change it into a better formula which can be far better than this demo.   In fact, even removing the mask to avoid computing loss function of loss_ii won't do harm to the final results.
